server/src/main/java/org/elasticsearch/health/stats/HealthApiStats.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.stats;
+
+import org.elasticsearch.common.metrics.Counters;
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.health.Diagnosis;
+import org.elasticsearch.health.GetHealthAction;
+import org.elasticsearch.health.HealthIndicatorResult;
+import org.elasticsearch.health.HealthStatus;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * This class tracks the health api calls and counts the statuses that have been encountered along with the unhealthy indicators and
+ * diagnoses. An example of what the stats look like is seen below. The stats are exposed via the xpack usage api for telemetry.
+ * {
+ *   "invocations": {
+ *     "total": 22,
+ *     "verbose_true": 12,
+ *     "verbose_false": 10
+ *   },
+ *   "statuses": {
+ *     "green": 10,
+ *     "yellow": 4,
+ *     "red": 8,
+ *     "values": ["green", "yellow", "red"]
+ *   },
+ *   "indicators": {
+ *     "red" : {
+ *       "master_stability": 2,
+ *       "ilm":2,
+ *       "slm": 4,
+ *       "values": ["master_stability", "ilm", "slm"]
+ *     },
+ *     "yellow": {
+ *       "disk": 1,
+ *       "shards_availability": 1,
+ *       "master_stability": 2,
+ *       "values": ["disk", "shards_availability", "master_stability"]
+ *     }
+ *   },
+ *   "diagnoses": {
+ *     "red": {
+ *       "elasticsearch:health:shards_availability:primary_unassigned": 1,
+ *       "elasticsearch:health:disk:add_disk_capacity_master_nodes": 3,
+ *       "values": [
+ *         "elasticsearch:health:shards_availability:primary_unassigned",
+ *         "elasticsearch:health:disk:add_disk_capacity_master_nodes"
+ *       ]
+ *     },
+ *     "yellow": {
+ *       "elasticsearch:health:disk:add_disk_capacity_data_nodes": 1,
+ *       "values": ["elasticsearch:health:disk:add_disk_capacity_data_nodes"]
+ *     }
+ *   }
+ * }
+ */
```

Comment:
Would this kind of documentation belong to the xpack usage response/representation ?

Reply:
Hm, I see.... This is closer to the usage output than anything else. I will move it